packages/orbit-components/src/tmp-tailwind/ButtonLink/index.tsx

```diff
@@ -0,0 +1,108 @@
+  secondary:
+    "bg-button-link-secondary-background hover:bg-button-link-secondary-background-hover active:bg-button-link-secondary-background-active focus:bg-button-link-secondary-background-focus text-button-link-secondary-foreground focus:text-button-link-secondary-foreground-focus active:text-button-link-secondary-foreground-active hover:text-button-link-secondary-foreground-hover active:shadow-button-active-pale",
+  critical:
+    "bg-button-link-critical-background hover:bg-button-link-critical-background-hover active:bg-button-link-critical-background-active focus:bg-button-link-critical-background-focus text-button-link-critical-foreground focus:text-button-link-critical-foreground-focus active:text-button-link-critical-foreground-active hover:text-button-link-critical-foreground-hover active:shadow-button-active",
+};
+
+const typeCompactStyles: Record<Type, string> = {
+  primary:
+    "text-button-link-primary-foreground focus:text-button-link-primary-foreground-focus active:text-button-link-primary-foreground-active hover:text-button-link-primary-foreground-hover",
+  secondary:
+    "text-button-link-secondary-foreground focus:text-button-link-secondary-compact-foreground-hover active:text-button-link-secondary-compact-foreground-active hover:text-button-link-secondary-compact-foreground-hover",
+  critical:
+    "text-button-link-critical-foreground focus:text-button-link-critical-foreground-focus active:text-button-link-critical-foreground-active hover:text-button-link-critical-foreground-hover",
+};
+
+const typeDisabledStyles: Record<Type, string> = {
+  primary: "bg-button-link-primary-background text-button-link-primary-foreground",
+  secondary: "bg-button-link-secondary-background text-button-link-secondary-foreground",
+  critical: "bg-button-link-critical-background text-button-link-critical-foreground",
+};
+
+const typeCompactDisabledStyles: Record<Type, string> = {
+  primary: "text-button-link-primary-foreground",
+  secondary: "text-button-link-secondary-foreground",
+  critical: "text-button-link-critical-foreground",
+};
+
+const ButtonLink = React.forwardRef<HTMLButtonElement, Props>(
+  (
+    {
+      type = "primary",
+      size = "normal",
+      compact,
+      disabled,
+      children,
+      iconLeft,
+      iconRight,
+      ...props
+    },
+    ref,
+  ) => {
+    return (
+      <ButtonPrimitive
+        ref={ref}
+        iconLeft={iconLeft}
+        iconRight={iconRight}
+        disabled={disabled}
+        {...props}
+        className={cx(
+          "space-x-xs rtl:space-x-reverse",
+          sizeStyles[size],
+          compact !== true && disabled !== true && typeStyles[type],
+          compact === true && disabled !== true && typeCompactStyles[type],
+          compact !== true && disabled === true && typeDisabledStyles[type],
+          compact === true && disabled === true && typeCompactDisabledStyles[type],
+          compact !== true &&
+            children != null &&
```

Comment:
this condition is repetitive, could you abstract that to a single variable instead? 🤔 

```
     compact !== true &&
            children != null &&
            iconLeft == null &&
            iconRight == null &&
```